src/Adyen/Service/ResourceModel/Checkout/Recurring.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Adyen\Service\ResourceModel\Checkout;
+
+class Recurring extends \Adyen\Service\AbstractResource
+{
+    /**
+     * @var string
+     */
+    protected $endpoint;
+
+    const STORED_PAYMENT_METHODS = '/storedPaymentMethods';
+
+    /**
+     * Include applicationInfo key in the request parameters
+     *
+     * @var bool
+     */
+    protected $allowApplicationInfo = true;
+
+
+    public function delete($recurringId, array $queryParams = [])
+    {
+        $url = $this->checkoutEndpoint . self::STORED_PAYMENT_METHODS . "/" . $recurringId;
+        return $this->requestHttp($url, 'delete', $queryParams);
```

Comment:
The ```$queryParams``` here won't actually work as the requestHttp in the curlclient only reads query params for a get request. Perhaps you can add a statement in the curlclient/requestHttp to also take query parameters if the request method is a delete.

Reply:
Good catch. Would there be any reason to keep checking for the http method?

Reply:
Yeah we do have to check it, because the params variable also get's used for body params in POST/PATCH. This is exactly why I completely copied and modified this whole method in my automation PR's.